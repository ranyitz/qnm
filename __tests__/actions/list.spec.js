const listAction = require('../../src/actions/list');
const { resolveWorkspace } = require('../utils');

describe('list', () => {
  it('should list the versions of mixed modules', () => {
    const workspace = resolveWorkspace('mix-modules');
    const output = listAction(workspace);

    expect(output).toMatchSnapshot();
  });

  it('should disable colors', () => {
    const workspace = resolveWorkspace('mix-modules');
    const output = listAction(workspace, {
      noColor: true,
    });

    expect(output).toMatchSnapshot();
  });

  it('should show why info', () => {
    const workspace = resolveWorkspace('single-module');
    const output = listAction(workspace, {
      why: true,
    });

    expect(output).toMatchSnapshot();
  });
});
