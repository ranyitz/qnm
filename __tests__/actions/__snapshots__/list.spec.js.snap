// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list should list the versions of mixed modules 1`] = `
"[4m@scope/test[24m
â””â”€â”€ 1.0.0

[4manother[24m
â””â”€â”€ 1.0.0

[4mtest[24m
â”œâ”€â”€ 1.0.0
â””â”€â”¬ [90manother[39m
  â””â”€â”€ 1.0.0
"
`;
